sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java

```diff
@@ -54,7 +54,6 @@ public static Object getValueByColumnType(final ResultSet resultSet, final int c
         ResultSetMetaData metaData = resultSet.getMetaData();
         switch (metaData.getColumnType(columnIndex)) {
             case Types.BIT:
-                return resultSet.getBytes(columnIndex);
```

Comment:
mybatis, postgresql jdbc driver,sql server jdbc driver handle BIT type to BOOLEAN, only MySQL BIT type can store a long value. sharding-sphere should convert the byte array to boolean if this method return byte array, otherwise cause #2604 .

Reply:
#2604 should assign to you? @tuohai666

Reply:
> mybatis, postgresql jdbc driver,sql server jdbc driver handle BIT type to BOOLEAN, only MySQL BIT type can store a long value. sharding-sphere should convert the byte array to boolean if this method return byte array, otherwise cause #2604 .

mybatis is not at the same level of jdbc driver. If mysql jdbc treat bit as a 8-bit integer, while mybatis treat bit as boolean, that means mybatis get wrong, are you sure?

Reply:
> #2604 should assign to you? @tuohai666

I don't think mybatis made such a simple mistake. This issue probably caused by misunderstanding or wrong usage.

Reply:
[TypeHandlerRegistry](https://github.com/mybatis/mybatis-3/blob/master/src/main/java/org/apache/ibatis/type/TypeHandlerRegistry.java#L69). Mybatis can correctly convert field classes because it looks for TypeHandlers based on field classes, but sharding-sphere converts fields based on JDBC types.

Reply:
> mybatis, postgresql jdbc driver,sql server jdbc driver handle BIT type to BOOLEAN, only MySQL BIT type can store a long value. sharding-sphere should convert the byte array to boolean if this method return byte array, otherwise cause #2604 .

There's no **byte array** concept in any aspect of the BIT type. The description is unclear.